src/haz3lcore/statics/TypBase.re

```diff
@@ -440,10 +440,21 @@ and Ctx: {
     kind: Kind.t,
   };
 
+  [@deriving (show({with_path: false}), sexp, yojson)]
+  type constructor_entry = {
+    name: Var.t,
+    id: Id.t,
+    typ: Typ.t,
+    /* Temporary variables. Better implementation is a TO-DO. */
+    nth: int,
```

Comment:
per other comment, we shouldn't have index calculations in the context because we have anonymous sums as well. we need to use the normalized type itself to determine indices wherever needed, and we should also be invariant to reordering (i.e. index is based on alphabetized position of constructor within the sum type rather than the list position)

Reply:
@pigumar1 there is still an index here -- what did you decide to do here?